### ISSUE TITLE
A profile for early-checking of templates

### ISSUE THREAD

**Issue by AbhinavK00** [BjarneStroustrup/profiles#2]
This is more in line with C++0x concept design and other languages such as Swift, Rust, Haskell etc. Now that C++ has concepts, it would be nice to have a profile that performs early definition checking to function and class templates. This suggestion is inspired from a discussion on cppfront github [here](https://github.com/hsutter/cppfront/discussions/591).
While I could not convince Herb to add early definition checking to cppfront, it was clear that the feature does have a place in today's C++.

**Comment by BjarneStroustrup** [comment]
Yes, that would fall into the realm of profiles. It is a restriction that does not change the semantics of a program that obeys it.
